src/arrayMethodJoin.js

```diff
@@ -1,11 +1,29 @@
 'use strict';
 
-/**
- * Implement method join
- */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedArray = '';
+    let checkedSeparator;
+
+    if (separator === null) {
+      checkedSeparator = 'null';
+    } else {
+      checkedSeparator = separator;
+    }
+
+    for (let element = 0; element < this.length; element++) {
+      if (this[element] === null || this[element] === undefined) {
+        this[element] = '';
+      }
+
+      if (element === this.length - 1) {
+        joinedArray += this[element];
+        continue;
+      }
+      joinedArray += this[element] + checkedSeparator;
+    }
+
+    return joinedArray;
```

Comment:
if you go backwards your code will be more readable
```suggestion
    for (let element = 0; element < this.length; element++) {
      if (this[element] !== null || this[element] !== undefined) {
        joinedArray += this[i];
      }

      if (element !== this.length - 1) {
        joinedArray += separator ;
      }
    }

    return joinedArray;
```